### PR TITLE
Django 1.10 compatibility

### PR DESCRIPTION
**Pull request by mgeist**
@BetterWorks/goals 
@terite 

Created a placeholder v1.3.0.3 branch which holds our existing release that we use, not sure why that's not in a branch atm. Created another v1.3.0.5 to hold this release, not sure why theres a 1.3.0.4 tag, but didn't want to step on anything.

Cherrypicked all the changes from the original repo, added a test change since excluded fields are handled differently in django 1.10

### CODE REVIEW

File: tests/core/test_bungiesearch.py [hunk 1]
```diff
@@ -291,8 +291,7 @@ def test_optimal_queries(self):
         db_item = NoUpdatedField.objects.get(pk=1)
         src_item = NoUpdatedField.objects.search.query('match', title='My title')[0]
         self.assertEqual(src_item.id, db_item.id, 'Searching for the object did not return the expected object id.')
-        self.assertTrue(src_item._meta.proxy, 'Was expecting a proxy model after fetching item.')
-        self.assertEqual(src_item._meta.proxy_for_model, NoUpdatedField, 'Proxy for model of search item is not "NoUpdatedField".')
+        self.assertEqual(src_item.get_deferred_fields(), {'description'})
```
Comment (mgeist) [inline tests/core/test_bungiesearch.py]: This is the only change I personally made, besides the version bump

**Comment by diwu1989** [comment]
Do you know why the upstream version seem to have tests passing, but this PR is failing on travis with errors like
`elasticsearch_dsl.exceptions.UnknownDslObject: DSL class `missing` does not exist in query.
`AttributeError: Could not find search alias named fields. Is this alias defined in BUNGIESEARCH["ALIASE`

**Comment by mgeist** [comment]
@diwu1989  I didn't look into those specifically as they were matching parity with the existing bungiesearch release on our fork, I want to as time permits but so far haven't had a chance.

**Comment by danielcmartin** [comment]
lgtm based on your commit and that its working on on your server

**Comment by terite** [comment]
The test failures I've investigated so far look like backwards incompatibilities with elasticsearch-dsl 5.0. The requirements are updated in upstream bungiesearch, and we're locked at a working version for our use cases.

LGTM

**Pull request merged**